src/transport/raw/TCP.cpp

```diff
@@ -324,11 +324,11 @@ CHIP_ERROR TCPBase::ProcessReceivedBuffer(Inet::TCPEndPoint * endPoint, const Pe
         {
             return err;
         }
-        uint16_t messageSize = LittleEndian::Get16(messageSizeBuf);
-        if (messageSize >= kMaxMessageSize)
+        uint32_t messageSize = LittleEndian::Get32(messageSizeBuf);
+        if (messageSize >= kMaxTCPMessageSize)
         {
-
-            // This message is too long for upper layers.
+            // Message is too big for node to process. Disconnect with peer.
+            CloseConnectionInternal(state, CHIP_ERROR_MESSAGE_TOO_LONG, SuppressCallback::No);
```

Comment:
Hmm, this is a behavior change that does not seem directly related to the "replace uint16_t with uint32_t" bits going on elsewhere here.  

And this behavior change was _definitely_ not in the PR description.  This change should probably be a separate PR with a clear explanation of why it's being made.